### PR TITLE
Adding Lambda Tutorial for OCI Images

### PR DESCRIPTION
**Pull request by davejson**
Description of changes:

Adding a tutorial for how Lambda OCI Images can be deployed using the ACK controller.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  - [Docker](https://docs.docker.com/engine/install/) - A tool to build, share, and run containers.
+
+### Install the ACK service controller for Lambda
+
+Log into the Helm registry that stores the ACK charts:
+```bash
+aws ecr-public get-login-password --region us-east-1 | helm registry login --username AWS --password-stdin public.ecr.aws
+```
+
+Deploy the ACK service controller for Amazon Lambda using the [lambda-chart Helm chart](https://gallery.ecr.aws/aws-controllers-k8s/lambda-chart). Resources should be created in the `us-east-1` region:
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:43]:
```suggestion
Deploy the ACK service controller for Amazon Lambda using the [lambda-chart Helm chart](https://gallery.ecr.aws/aws-controllers-k8s/lambda-chart). This example creates resources in the `us-east-1` region, but you can use any other region supported in AWS.
```

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+## Create Lambda function handler
+The Lambda [function handler](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html) is the method in your function code that processes events. When your function is invoked, Lambda runs the handler method.
+
+```bash
+cat <<EOF > app.js
+exports.handler = async (event) => {
+    const response = {
+        statusCode: 200,
+        body: JSON.stringify('Hello from Lambda!'),
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:66]:
```suggestion
        body: JSON.stringify('Hello from Lambda!')
```

(My JavaScript knowledge may be fading, but IIRC you're not supposed to have extra commas in object definitions).

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+EOF
+```
+Install dependencies and build a Docker image called hello-world.
+
+```shell
+npm init -y
+docker build -t hello-world .
+```
+## Publish the Docker image to ECR
+Publish the Docker image to an ECR repository. It's a requirement for container images to be published to the ECR repository to run Lambda OCI image functions.
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:94]: Double checking if using the ECR repository is required for this or if it's just recommended for the tutorial.

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+```shell
+export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query "Account" --output text)
+export AWS_REGION=us-west-2
+
+aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin ${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:100]: Two things:

1. Is there a reason why the tutorial switches to `us-west-2` here when we talk about setting up the controller for `us-east-1`?
2. There is mixed use of shell variables and hard coded strings here, e.g. `us-west-2` is spelled out in the beginning.
3. I recommend wrapping the strings that have variables, i.e. `"${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com"`

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+aws ecr create-repository --repository-name hello-world --image-scanning-configuration scanOnPush=true --image-tag-mutability MUTABLE
+docker tag  hello-world:latest ${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com/hello-world:latest
+docker push ${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com/hello-world:latest
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:103]: Wrap strings that have variables.

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+```
+
+## Deploy the Lambda OCI function using the ACK Lambda controller
+Execute the following commands to create a manifest containing the Lambda OCI function and submit this manifest to the EKS cluster using kubectl.
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:107]:
```suggestion
The following example creates a manifest that contains the Lambda OCI function. It then uses `kubectl` to create the resource in Kubernetes:
```

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+function.lambda.services.k8s.aws/lambda-oci-ack created
+```
+To get details about the Lambda function, run the following.
+
+```bash
+kubectl describe function/${FUNCTION_NAME}
+```
+
+## Invoke the Lambda OCI Function
+After you have verified that the Lambda OCI function is deployed correctly, you can now invoke your function through the [AWS CLI](https://docs.aws.amazon.com/cli/latest/reference/lambda/index.html).
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:147]:
```suggestion
After you have verified that the Lambda OCI function is deployed correctly, you can invoke the function through the [AWS CLI](https://docs.aws.amazon.com/cli/latest/reference/lambda/index.html).
```

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+```
+You should get a confirmation that the function was created successfully.
+
+```
+function.lambda.services.k8s.aws/lambda-oci-ack created
+```
+To get details about the Lambda function, run the following.
+
+```bash
+kubectl describe function/${FUNCTION_NAME}
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:143]:
```suggestion
kubectl describe "function/${FUNCTION_NAME}"
```

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+## Next steps
+
+The ACK service controller for Amazon Lambda is based on the [Amazon Lambda API](https://docs.aws.amazon.com/lambda/latest/dg/API_Reference.html).
+
+Refer to [API Reference](https://aws-controllers-k8s.github.io/community/reference/) for *Lambda* to find
+all the supported Kubernetes custom resources and fields.
+
+### Cleanup
+
+You can delete your Lambda OCI function using the `kubectl delete` command.
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:168]:
```suggestion
You can delete your Lambda OCI function using the `kubectl delete` command:
```

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+        statusCode: 200,
+        body: JSON.stringify('Hello from Lambda!'),
+    };
+    return response;
+};
+EOF
+```
+
+## Create and Build a Docker Image
+Create a Dockerfile to build an image from.
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:74]:
```suggestion
Create a Dockerfile that will be used to build the image that will be used for our Lambda function:
```

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+FROM public.ecr.aws/lambda/nodejs:14
+
+COPY app.js package.json ./
+
+RUN npm install
+
+CMD [ "app.handler" ]
+EOF
+```
+Install dependencies and build a Docker image called hello-world.
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:87]:
```suggestion
Build the Docker image in your local environment. You will need to install dependencies using `npm`:
```

**Review by jkatz — CHANGES_REQUESTED** [review]

File: docs/content/docs/tutorials/lambda-oci-example.md (hunk 3 above)
Comment (davejson) [reply docs/content/docs/tutorials/lambda-oci-example.md:94]: Yes, its required

File: docs/content/docs/tutorials/lambda-oci-example.md (hunk 4 above)
Comment (davejson) [reply docs/content/docs/tutorials/lambda-oci-example.md:100]: Good catch. Will redefine.

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+Log into the Helm registry that stores the ACK charts:
+
+```bash
+aws ecr-public get-login-password --region us-west-2 | helm registry login --username AWS --password-stdin public.ecr.aws
+```
+
+Deploy the ACK service controller for Amazon Lambda using the [lambda-chart Helm chart](https://gallery.ecr.aws/aws-controllers-k8s/lambda-chart). This example creates resources in the `us-west-2` region, but you can use any other region supported in AWS.
+
+```bash
+export RELEASE_VERSION=`curl -sL https://api.github.com/repos/aws-controllers-k8s/$SERVICE-controller/releases/latest | grep '"tag_name":' | cut -d'"' -f4`
```
Comment (jkatz) [inline docs/content/docs/tutorials/lambda-oci-example.md:47]:
```suggestion
SERVICE=lambda
RELEASE_VERSION=$(curl -sL "https://api.github.com/repos/aws-controllers-k8s/${SERVICE}-controller/releases/latest" | grep '"tag_name":' | cut -d'"' -f4)
```

**Review by jkatz — CHANGES_REQUESTED** [review]
Looks great! One small change.

**Review by jkatz — APPROVED** [review]

File: docs/content/docs/tutorials/lambda-oci-example.md [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+
+```bash
+kubectl describe "function/${FUNCTION_NAME}"
+```
+
+## Invoke the Lambda OCI Function
+After you have verified that the Lambda OCI function is deployed correctly, you can invoke the function through the [AWS CLI](https://docs.aws.amazon.com/cli/latest/reference/lambda/index.html).
+
+```bash
+aws lambda invoke --function-name ${FUNCTION_NAME} response.json
```
Comment (A-Hilaly) [inline docs/content/docs/tutorials/lambda-oci-example.md:153]: tiny nit: you can also use `aws lambda invoke --function-name --region us-west-2 demo-function-list-items /dev/stdout | jq`

**Review by A-Hilaly — APPROVED** [review]
This is great! Thanks @davejson!

**Comment by A-Hilaly** [comment]
👍 
/approve
/lgtm

**Pull request merged**